scriptlets/install_checkbox_snaps

```diff
@@ -10,77 +10,107 @@
-[[ "$FRONTEND_TRACK" == uc* ]] && STRICT_FRONTEND=true || STRICT_FRONTEND=false
 
-if $STRICT_FRONTEND; then
-  echo "Frontend is a strict snap, using --devmode"
-  EXTRA_SNAP_INSTALL_FLAG="--devmode"
-else
-  echo "Frontend is a classic snap, using --classic"
-  EXTRA_SNAP_INSTALL_FLAG="--classic"
+# install the frontend snap as a strict snap and revert to classic if that fails
+# (strict snaps need a different install flag and the connections to be made)
+echo "Installing frontend snap: $FRONTEND_NAME from $FRONTEND_CHANNEL (as a strict snap, using --devmode)"
+STRICT_FRONTEND=true
+_run sudo snap install --no-wait $FRONTEND_NAME --devmode --channel=$FRONTEND_CHANNEL
+if [ "$?" -ne 0 ]; then
+    echo "Failed to install $FRONTEND_NAME as a strict snap"
+    echo "Installing frontend snap: $FRONTEND_NAME from $FRONTEND_CHANNEL (as a classic snap, using --classic)"
+    STRICT_FRONTEND=false
+    _run sudo snap install --no-wait $FRONTEND_NAME --classic --channel=$FRONTEND_CHANNEL
 fi
-
-_run sudo snap install --no-wait $FRONTEND_NAME $EXTRA_SNAP_INSTALL_FLAG --channel=$FRONTEND_TRACK/$RISK
 wait_for_snap_changes
 
+# run the configure hook of the frontend snap
+_run sudo snap set $FRONTEND_NAME agent=enabled
+# note: setting "slave" instead of "agent" has been deprecated but is still necessary for some snaps
+_run sudo snap set $FRONTEND_NAME slave=enabled
+
 if $STRICT_FRONTEND; then
-  # these plugs are not automatically connected but are necessary to run some
-  # tests. Namely:
-  set +e
-  # Here we are handling results on a per-interface basis because we may not need
-  # these interfaces as we may not need to run the tests they are needed for
-
-  # Network tests
-  _run sudo snap connect $FRONTEND_NAME:network-manager network-manager:service
-  if [ $? -ne 0 ]; then
-    echo "Failed to connect plug. Networking (mostly wifi) tests may not work"
-  fi
-  # bluetooth tests
-  _run sudo snap connect $FRONTEND_NAME:bluez bluez:service
-  if [ $? -ne 0 ]; then
-    echo "Failed to connect plug. Bluetooth tests may not work"
-  fi
-  _run sudo snap connect $FRONTEND_NAME:bluetooth-control core:bluetooth-control
-  if [ $? -ne 0 ]; then
-    echo "Failed to connect plug. Bluetooth tests may not work"
-  fi
-  set -e
-
-  # force connect also the default plugs. This is necessary on some machines but
-  # is effectively a no-op when it would not be needed
-  _run sudo snap connect $FRONTEND_NAME:checkbox-runtime $RUNTIME_NAME:checkbox-runtime
-  _run sudo snap connect $FRONTEND_NAME:provider-certification-client $RUNTIME_NAME:provider-certification-client
-  _run sudo snap connect $FRONTEND_NAME:provider-checkbox $RUNTIME_NAME:provider-checkbox
-  _run sudo snap connect $FRONTEND_NAME:provider-resource $RUNTIME_NAME:provider-resource
+    # these are the default plugs (although some custom frontends may not have them)
+    CONNECTIONS="checkbox-runtime provider-checkbox provider-resource provider-certification-client"
```

Comment:
why don't we use the same line as below to fetch the connections that the runtime support? I'm pretty sure there are more

Reply:
Oh, so you mean match the frontend plugs to the runtime slots? (instead of using a predefined set)

Reply:
I have now generalised the way the frontend plugs connect to both runtime slots, as well as other slots available for matching interfaces. No plugs or slots are now hard-coded.